Tribler/community/tunnel/tunnel_community.py

```diff
@@ -148,7 +148,6 @@ def on_ip_address(ip_address):
                     except Exception as exception:
                         self.tunnel_logger.error("Failed to write data to transport: %s. Destination: %r",
                                                  exception, destination)
-                        raise
```

Comment:
1: You fail to write the data but still increase the counter?
2: That's a catchAll try/except block.
3: That should be squashed with the commit where you refactored this.